src/shared/ask-password-api.c

```diff
@@ -113,6 +114,26 @@ static int touch_ask_password_directory(AskPasswordFlags flags) {
         return 1; /* did something */
 }
 
+static usec_t keyring_timeout(void) {
+        static usec_t saved_timeout = USEC_INFINITY;
+        int r;
+
+        if (saved_timeout != USEC_INFINITY)
```

Comment:
hmm, we should allow infinite caching I think. And we should allow disabling of caching. hence you probably can't use neither 0 nor USEC_INFINITY as niche value for "unset"...

probably need to pair it with a boolean